Feature/search by day time

yanyuzhong opened the pull request:
![image](https://user-images.githubusercontent.com/48166924/88467188-0b69f400-ce9a-11ea-9995-7c7d39498b3f.png)
![image](https://user-images.githubusercontent.com/48166924/88467213-3eac8300-ce9a-11ea-97b0-9ab83cd275d5.png)

TheWillMundy commented on client/src/components/search/CourseSearch.js:
```diff
@@ -6,10 +6,27 @@ import { useQuery, gql } from "@apollo/client";
 
 const dummy = { label: "", value: "" };
 
+const ITEM_HEIGHT = 48;
+const ITEM_PADDING_TOP = 8;
+const MenuProps = {
+    PaperProps: {
+        style: {
+            maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
+            width: 250,
+        },
+    },
+};
+
```
Specify in a comment that these are used for the Select component below (bc otherwise someone might try to refactor into CSS)

TheWillMundy commented:
After those minor changes / responses, this will be good to approve. Overall it looks awesome, nice work!

yanyuzhong commented:
I forgot to pull whatever's in the develop branch. So I refactored my code and merge it with the develop branch. I've also made the course day search bar to be the same as the other search options. And I deleted the unused code in the backend. The main changes are in CourseSearch.js, CompiledLists.js, and CourseList.js. 
![image](https://user-images.githubusercontent.com/48166924/88492034-5274e980-cf6d-11ea-821f-2c9804644fce.png)
![image](https://user-images.githubusercontent.com/48166924/88492094-ee9ef080-cf6d-11ea-86fe-38676ea62925.png)
![image](https://user-images.githubusercontent.com/48166924/88492117-05ddde00-cf6e-11ea-9daa-837ad2765cc6.png)

TheWillMundy commented on client/nwb.config.js:
```diff
@@ -3,7 +3,8 @@ module.exports = {
         define: {
             "process.env.REACT_APP_SERVICE_URL": JSON.stringify(
-                "https://develop-dot-ultrascheduler.uc.r.appspot.com/auth"
+                "http://localhost:3001/auth"
+                // "https://develop-dot-ultrascheduler.uc.r.appspot.com/auth"
```
We need the other URL for production; could you switch it back?

TheWillMundy commented on client/src/apollo.js:
```diff
@@ -28,8 +28,8 @@ const authLink = setContext((_, { headers }) => {
-    // uri: "http://localhost:3000/graphql",
-    uri: "/graphql"
+    uri: "http://localhost:3000/graphql",
+    // uri: "/graphql",
```
Same thing as above here.

TheWillMundy commented on client/src/components/search/Selection.js:
```diff
@@ -18,15 +18,32 @@ const style = {
-    return <div></div>;
-  }
+    if (show) {
+        // if (selected !== "") {
+        //     return (
+        //         <div style={style}>
+        //             <Select
+        //                 value={selected}
+        //                 onChange={handleChange}
+        //                 options={options}
+        //                 isMulti={true}
+        //             />
+        //         </div>
+        //     );
+        // }
```
Do we need this still?
